__tests__/app.test.js

```diff
@@ -40,25 +41,99 @@ describe('GET /api', () => {
+            .get('/api/articles/3')
+            .expect(200)
+            .then((response) => {
+                expect(response.body.article).toEqual(
+                {
+                    article_id: 3,
+                    title: "Eight pug gifs that remind me of mitch",
+                    topic: "mitch",
+                    author: "icellusedkars",
+                    body: "some gifs",
+                    created_at: '2020-11-03T09:12:00.000Z',
+                    votes: 0,
+                })
+
+            })
+            
+        });
+        test('400: Given an invalid data-type, send the client a 400 bad request.', () => {
+            return request(app)
+            .get('/api/articles/silly')
+            .expect(400)
+            .then((response) => {
+                expect(response.body.msg).toEqual('Bad Request')
+            })
+        })
+        test('404: Valid but non-existent ID', () => {
+            return request(app).get('/api/articles/999999')
+            .expect(404)
+            .then((response) => {
+                expect(response.body.msg).toEqual('Not Found')
+            })
+        })
+    });
+
+    describe('PATCH /api/articles/:article_id', () => {
+        test('201: Should update the relevant property by the given value of the specified id.', () => {
+
+            const updateVote = { inc_votes: 40 } 
+
+            return request(app)
+            .patch('/api/articles/3')
+            .expect(201)
+            .send(updateVote)
+            .then((response) => {
+                expect(response.body.update).toEqual(
+                    {
+                        article_id: 3,
+                        title: "Eight pug gifs that remind me of mitch",
+                        topic: "mitch",
+                        author: "icellusedkars",
+                        body: "some gifs",
+                        created_at: '2020-11-03T09:12:00.000Z',
+                        votes: 40,
+                    })
+
+            })
             
-    //         return request(app)
-    //         .get('/api/articles/3')
-    //         .expect(200)
-    //         .then((response) => {
-    //             ex
+        });
+        test('400: Given a malformed body, respond with a 400: bad request',() => {
+            const badBody = {}
+            return request(app)
+            .patch('/api/articles/3')
+            .expect(400)
+            .send(badBody)
+            .then((response) => {
+                expect(response.body.msg).toEqual('Bad Request')
 
-    //         })
+            })
+        })
+        test('400: Given an incorrect data type, response with a 400: bad request ', () => {
+            const wrongData = { inc_votes: 'vote' }
+
+            return request(app)
+            .patch('/api/articles/3')
+            .expect(400)
+            .send(wrongData)
+            .then((response) => {
+                expect(response.body.msg).toEqual('Bad Request')
+
```

Comment:
Would like to see a test for trying to patch an article that doesnt yet exist, and a test for an article_id that is not valid (not a number)